src/main/scala/chisel3/util/experimental/BoringUtils.scala

```diff
@@ -224,9 +224,14 @@ object BoringUtils {
     else if (DataMirror.hasOuterFlip(source)) Flipped(chiselTypeOf(source))
     else chiselTypeOf(source)
     def purePortType = createProbe match {
-      case Some(pi) if pi.writable => RWProbe(purePortTypeBase)
-      case Some(pi)                => Probe(purePortTypeBase)
-      case None                    => purePortTypeBase
+      case Some(pi) =>
+        // If the source is already a probe, don't double wrap it in a probe.
+        purePortTypeBase.probeInfo match {
```

Comment:
Things get strange if the bore/tap target is mixed probe/non-probe, but that's already true.

(or if has incompatible probe type vs requested kind)

Not sure what the errors look like if this is wrong.

Reply:
Not something to be addressed here unless it seems easy/reasonable to address while visiting (?), just a note.